src/com/gfredericks/debug_repl/async.clj

```diff
@@ -0,0 +1,65 @@
+(ns com.gfredericks.debug-repl.async
+  (:require
+    [com.gfredericks.debug-repl :as debug-repl]
+    [com.gfredericks.debug-repl.util :as util])
+  (:import (java.util.concurrent
+            ArrayBlockingQueue
+            TimeUnit)))
+
+(def the-executor nil) ;; global mutable state
+
+(def ^:private msg-var
+  (if (util/require? 'nrepl.server)
+    (resolve 'nrepl.middleware.interruptible-eval/*msg*)
+    (resolve 'clojure.tools.nrepl.middleware.interruptible-eval/*msg*)))
+
+(defn can-break?
+  "Implementation detail. Subject to change."
```

Comment:
why not just make it actually `^:private`?

Reply:
Because it's used from the macro below

Reply:
I guess `msg-var` can't be private either :/

Reply:
It can, because it's called by a public function, not by the macro directly.

Macros can't generate code with direct references to private vars (I assume because they're not available in the call-site).

Reply:
augh okay right, I'm just reading sloppily, sorry; I was imagining that `can-break?` could be inlined, at which point `msg-var` couldn't be private; so I guess it's one or the other